torch/fx/passes/reinplace.py

```diff
@@ -0,0 +1,257 @@
+            if not isinstance(arg, Iterable):
+                arg = [arg]
+            for a in arg:
+                if isinstance(a, fx.node.Node):
+                    # We map to the actual fx.Node object, because later we might need to
+                    # modify the arguments that this node is called with
+                    self.names_to_used_nodes[a.name].append([self.node_counter, node])
+
+        if node.op == 'placeholder':
+            # Update alias map + input names when we see a placeholder
+            r = super().placeholder(node.target, args, kwargs)
+            self.input_names.append(node.name)
+            self.alias_map[StorageWeakRef(r.storage()).cdata].add(node.name)
+            return r
+        elif node.op == 'call_function':
+            # Update alias map when we see a call_function
+            for runtime_arg, node_arg in itertools.chain(
+                    zip(args, node.args), zip(kwargs.values(), node.kwargs.values())):
+                if isinstance(runtime_arg, torch.Tensor):
+                    self.alias_map[StorageWeakRef(runtime_arg.storage()).cdata].add(node_arg.name)
+
+            # And update our mapping for variables that are copied to.
+            if node.target is torch.ops.aten.copy_.default:
+                self_arg = args[0]
+                self.last_node_where_variable_is_overwritten[StorageWeakRef(self_arg.storage()).cdata] = self.node_counter
+
+            return super().call_function(node.target, args, kwargs)
+        else:
+            return super().run_node(node)
+
+
+    def __call__(self, *args, **kwargs):
+        # Hmm, seems like fx.Interpreter.run() doesn't accept kwargs?
+        super().run(*args)
+        # Do some post-processing on our bookkeeping.
+        # We want to avoid returning data structures that involve StorageImpl,
+        # since that isn't the same from run to run, and replace it with variable names from the graph
+
+        # For the alias map, map [variable name] -> [names of all aliased variables]
+        alias_map = {}
+        for aliased_var_names in self.alias_map.values():
+            for v in aliased_var_names:
+                alias_map[v] = aliased_var_names
+
+        # For the copy_() map [program input var name] -> [node index of the last copy_() that wrote to the input's data]
+        last_node_where_input_alias_is_overwritten = defaultdict(lambda: -1)
+        for storage_ref, copy_node_idx in self.last_node_where_variable_is_overwritten.items():
+            # We only actually care about variables used in copy_() ops that are program inputs (or their aliases)
+            aliased_inputs = [inpt_name for inpt_name in self.input_names if inpt_name in self.alias_map[storage_ref]]
+            for inpt_name in aliased_inputs:
+                last_node_where_input_alias_is_overwritten[inpt_name] = copy_node_idx
+
+        return self.input_names, alias_map, self.names_to_used_nodes, last_node_where_input_alias_is_overwritten
```

Comment:
@Chillee so, if I understand you correctly, idiomatic way to implement this would be to FakeTensorProp and then use the fake tensor annotations on nodes instead of reinterpreting, right?